convex/lp/simplex.go

```diff
@@ -443,17 +447,21 @@ func initializeFromBasic(ab *mat64.Dense, b []float64) (xb []float64, err error)
 	return xb, nil
 }
 
-// extractColumns creates a new matrix out of the columns of A specified by cols.
-// TODO(btracey): Allow this to take a receiver.
-func extractColumns(A mat64.Matrix, cols []int) *mat64.Dense {
-	r, _ := A.Dims()
-	sub := mat64.NewDense(r, len(cols), nil)
+// extractColumns copies the columns specified by cols into the columns of dst.
+func extractColumns(dst *mat64.Dense, A mat64.Matrix, cols []int) {
+	r, c := dst.Dims()
+	ra, _ := A.Dims()
+	if ra != r {
+		panic("simplex: row mismatch")
```

Comment:
My thought was that this panic should never be triggered -- it can only be accessed internally. Thus, the panic purely belongs to simplex.